### ISSUE TITLE
Disallow use of angular functions

### ISSUE THREAD

**Issue by remcohaszing** [Gillespie59/eslint-plugin-angular#273]
There are rules which enforce the use of AngularJS functions.

**Good**

``` js
angular.isUndefined(foo);
```

**Bad**

``` js
foo === undefined;
```

I think these should all have an opposite variant as well:

**Config**

``` yaml
rules:
  angular/definedundefined:
  - 2
  - never
```

**Good**

``` js
foo === undefined;
```

**Bad**

``` js
angular.isUndefined(foo);
```

I figured the value of this extra property can be `always` or `never`, because this matches a lot of builtin ESLint rules.

**Comment by Gillespie59** [comment]
- [definedundefined] 
- [typecheck-array]
- [typecheck-date]
- [typecheck-function]
- [typecheck-number]
- [typecheck-object]
- [typecheck-regexp]
- [typecheck-string]

**Comment by himdel** [comment]
Anyone working on this? I could use this feature so I'm willing to take it :)
